lib/matplotlib/image.py

```diff
@@ -852,8 +852,8 @@ def make_image(self, magnification=1.0):
         l, b, r, t = self.axes.bbox.extents
         width = (round(r) + 0.5) - (round(l) - 0.5)
         height = (round(t) + 0.5) - (round(b) - 0.5)
-        width = width * magnification
-        height = height * magnification
+        width = int(width * magnification)
+        height = int(height * magnification)
```

Comment:
Looks good to me.  The only question it raises in my mind is whether after multiplication by magnification, a rounding operation would be better than truncation.